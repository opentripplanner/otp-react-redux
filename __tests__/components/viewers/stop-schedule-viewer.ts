import '../../test-utils/mock-window-url'
import {
  getMockInitialState,
  mockWithProvider
} from '../../test-utils/mock-data/store'
import { restoreDateNowBehavior, setDefaultTestTime } from '../../test-utils'
import StopScheduleViewer from '../../../lib/components/viewers/stop-schedule-viewer'

describe('components > viewers > stop viewer', () => {
  afterEach(restoreDateNowBehavior)
  beforeEach(setDefaultTestTime)

  it('should render with initial stop id and no stop times', () => {
    const mockState = getMockInitialState()
    mockState.otp.ui.viewedStop = {
      stopId: 'TriMet:13170'
    }

    expect(
      mockWithProvider(StopScheduleViewer, {}, mockState).snapshot()
    ).toMatchSnapshot()
  })
})
