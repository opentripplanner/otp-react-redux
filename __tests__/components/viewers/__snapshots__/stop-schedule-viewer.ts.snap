// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > viewers > stop viewer should render with initial stop id and no stop times 1`] = `
<div>
  <div
    class="stop-viewer base-color-bg"
  >
    <div
      class="stop-viewer-header"
    >
      <div
        class="back-button-container"
      >
        <button
          class="btn btn-sm btn-default"
          type="button"
        >
          <div
            style="display: contents;"
          >
            <span
              class="sc-jifHHV sc-hkwmXC frJUIW gWGcte"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-sc-ea9ulj-0 fqiNOa"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H109.3l105.3-105.4c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              common.forms.back
            </span>
          </div>
        </button>
      </div>
      <div
        class="sc-hJxDiT dbjOgr"
      >
        <div
          class="sc-jlIlqL sdJoA"
        >
          <h1>
            components.StopViewer.loadingText
          </h1>
        </div>
      </div>
      <div
        style="clear: both;"
      />
    </div>
  </div>
</div>
`;
